modules/apps/portal-search-solr8/portal-search-solr8-impl/src/main/java/com/liferay/portal/search/solr8/internal/query/BooleanQueryTranslatorImpl.java

```diff
@@ -30,32 +34,43 @@ public org.apache.lucene.search.Query translate(
 		BooleanQuery booleanQuery,
 		QueryVisitor<org.apache.lucene.search.Query> queryVisitor) {
 
-		List<BooleanClause<Query>> clauses = booleanQuery.clauses();
-
-		if (clauses.isEmpty()) {
-			return null;
-		}
-
-		org.apache.lucene.search.BooleanQuery.Builder builder =
+		org.apache.lucene.search.BooleanQuery.Builder booleanQueryBuilder =
 			new org.apache.lucene.search.BooleanQuery.Builder();
 
+		List<BooleanClause<Query>> clauses = booleanQuery.clauses();
+
 		for (BooleanClause<Query> booleanClause : clauses) {
 			org.apache.lucene.search.Query query = translate(
 				booleanClause.getClause(), queryVisitor);
 
 			if (query != null) {
-				builder.add(
+				booleanQueryBuilder.add(
 					query, translate(booleanClause.getBooleanClauseOccur()));
 			}
 		}
 
-		org.apache.lucene.search.Query query = builder.build();
+		BooleanFilter booleanFilter = booleanQuery.getPreBooleanFilter();
```

Comment:
I know you didn't create this API, but im just curious if you have any idea why a Boolean Query has a "Pre Boolean Filter"?

Maybe `BooleanQuery extends Query` was a mistake...